pkg/pillar/types/patchenvelopestypes.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type PatchEnvelopes struct {
+	AppsToEnvelopes map[string][]PatchEnvelopeInfo
+}
+
+// Key for pubsub
+func (PatchEnvelopes) Key() string {
+	return "zedagent"
+}
+
+func NewPatchEnvelopes() *PatchEnvelopes {
+	pe := &PatchEnvelopes{}
+	pe.AppsToEnvelopes = make(map[string][]PatchEnvelopeInfo)
+
+	return pe
+}
+
+func (pe *PatchEnvelopes) Add(peInfo PatchEnvelopeInfo, allowedApps []string) error {
+	for _, app := range allowedApps {
+		pe.AppsToEnvelopes[app] = append(pe.AppsToEnvelopes[app], peInfo)
+	}
+	return nil
+}
+
+func (pe PatchEnvelopes) Get(appUuid string) []PatchEnvelopeInfo {
+	return pe.AppsToEnvelopes[appUuid]
+}
+
+// PatchEnvelopeInfo - information
+// about patch envelopes
+type PatchEnvelopeInfo struct {
+	PatchId     string       `json:"patch-id"`
+	BinaryBlobs []BinaryBlob `json:"binary-blobs"`
+}
+
+func GetZipArchive(root string, pe PatchEnvelopeInfo) (string, error) {
```

Comment:
GetZipArchive should go into `pillar/utils`, `types` package should preferably contain only global type declarations and their methods.

Reply:
We can change signature to :D 
```go
func (pe PatchEnvelopeInfo) GetZipArchive(root string) (string, error) {
//...
}
```

Reply:
FWIW changing the signature makes sense; in any case this operators on a PE.

Reply:
@uncleDecart It is not about the signature. Package `types` should only contain structures defining APIs between pillar microservices. This is imported by every microservice, including "mmagent" in the future. Having this method here will add dependencies and increase binary size (which is unnecessary for everything outside of zedbox, i.e. edgeview, mmagent, etc.).

Reply:
Moved this code to utils